drivers: i2c: stm32: cleanup build issue if no I2C ports are configured

galak opened the pull request:
We can enable the driver, but not actually have any I2C ports enabled so
we will get build issues in this case.  Put ifdef protection around the
code so that we only build it if one of the ports is enabled.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ydamigos reviewed (commented):
What about the case that someone forgets to enable a I2C port? He will get no error and he will spend more time debugging the issue. If someone does not need the driver, he should disable it.

galak commented:
Well the port enablement should eventually be driven by DT, so we really shouldn't have that issue.

galak commented on drivers/i2c/i2c_ll_stm32.c:
```diff
@@ -17,6 +17,9 @@
 #define SYS_LOG_LEVEL CONFIG_SYS_LOG_I2C_LEVEL
 #include <logging/sys_log.h>
 
+#if defined(CONFIG_I2C_0) || defined(CONFIG_I2C_1) || \
```
Thanks, fixed.

mnkp commented:
@ydamigos made a good point. The change hides the problem and makes it more difficult to debug the issue. The compiler warning in case no ports are enabled is very much in place and helpful. We would better make it more not less explicit. E.g. by having something like the following check at the top of the file:
```
#if !(defined(CONFIG_I2C_1) || defined(CONFIG_I2C_2) || defined(CONFIG_I2C_3))
#error "No I2C port is enabled"
#endif
```
as en error or warning. If we don't want the warning it's better to leave the code the way it is now / abandon this PR.

galak commented:
The compile error breaks testsuite runs.  As I mentioned once we put the DT stuff in, this will not mater, as we'll use the DT configure a port being enabled or not.

mnkp commented:
But does it not mean that we configured the testcase incorrectly? So we should rather fix the testcase.

galak committed:
i2c: stm32: Cleanup how we enable the specific I2C driver

Right now we allow for the I2C subsystem to be built without any drivers
enabled that utilize it.  When we added support for the new STM32 I2C
driver we forced the I2C driver to be enabled if the I2C subsystem was
enabled.  While this makes a reasonable amount of sense, it breaks
current assumptions for various testcases that we need to cleanup.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
Yeah, so I looked into this and worked up a different solution.  We should fix the testcases, but will look at that in the future.
